modules/tinymce/src/core/main/ts/caret/CaretUtils.ts

```diff
@@ -170,19 +167,21 @@ const before = Fun.curry(beforeAfter, true) as (node: Node) => Range;
 const after = Fun.curry(beforeAfter, false) as (node: Node) => Range;
 
 const normalizeRange = (direction: number, root: Node, range: Range): Range => {
-  let node, container, location;
+  let node: Node;
   const leanLeft = Fun.curry(lean, true, root);
   const leanRight = Fun.curry(lean, false, root);
 
-  container = range.startContainer;
+  let container = range.startContainer;
   const offset = range.startOffset;
 
   if (CaretContainer.isCaretContainerBlock(container)) {
     if (!isElement(container)) {
       container = container.parentNode;
     }
 
-    location = container.getAttribute('data-mce-caret');
+    // TODO: The `isCaretContainerBlock` function already asserts container is an element, so
+    //  the above isElement check likely isn't needed which would remove the need for the cast.
```

Comment:
Tiny thing but I don't think this extra space was deliberate?

```suggestion
    // TODO: The `isCaretContainerBlock` function already asserts container is an element, so
    // the above isElement check likely isn't needed which would remove the need for the cast.
```

Reply:
Yes, it's deliberate otherwise IDEs, etc... assume it's a newline and not part of the TODO statement. We could move it over further so it's inline with the TODO if you prefer? eg:
```suggestion
    // TODO: The `isCaretContainerBlock` function already asserts container is an element, so
    //       the above isElement check likely isn't needed which would remove the need for the cast.
```